business/checkers/checker.go

```diff
@@ -1,11 +1,95 @@
 package checkers
 
-import "github.com/kiali/kiali/models"
+import (
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+)
 
-type Checker interface {
+type ObjectChecker interface {
+	GetType() string
+	GetGroupCheckers() []GroupChecker
+	GetIndividualCheckers(object kubernetes.IstioObject) []IndividualChecker
+	GetObjects() []kubernetes.IstioObject
+	Check() models.IstioValidations
+}
+
+type IndividualChecker interface {
 	Check() ([]*models.IstioCheck, bool)
 }
 
 type GroupChecker interface {
 	Check() models.IstioValidations
 }
+
+type ObjectCheck struct{}
```

Comment:
I guess you could remove all these Getters (GetType, GetGroupCheckers etc.) and have the corresponding stuff declared as struct fields instead. They would be provided at construction, for instance:

```
func NewDestinationRulesChecker() {
  return DestinationRulesChecker{
    Type: "destinationrule",
    // ...
  }
}
```

(you can pass lambdas as well)

Forget OOP ;-)

Reply:
It seems that I haven't learnt that lesson yet, hahaha

Reply:
@xeviknal You are not alone. It's also hard for me to stop thinking in a OOP way.
Then I read this: https://medium.com/@cscalfani/goodbye-object-oriented-programming-a59cda4c0e53 haha I've had almost all problems described in that article.